### PR TITLE
Zc/actor fk inversion migration

### PR DESCRIPTION
**Pull request by corps**
Preparing columns for actor foreign key inversion, extracted `is_dangerous=True` from https://github.com/getsentry/sentry/pull/46171

### CODE REVIEW

**Comment by corps** [comment]
Ok, I see, I need to further split this out -- there needs to be the migration that adds the columns, minus the indexes, then a follow up migration.  refactoring this a bit.

File: src/sentry/models/actor.py [hunk 1]
```diff
@@ -148,7 +173,10 @@ def resolve(self) -> Union["Team", "RpcUser"]:
         return fetch_actor_by_id(self.type, self.id)
 
     def resolve_to_actor(self) -> Actor:
-        return Actor.objects.get(id=self.resolve().actor_id)
+        obj = self.resolve()
+        if obj.actor_id is None and isinstance(obj, RpcUser):  # This can happen for users now.
```
Comment (markstory) [inline src/sentry/models/actor.py:177]: I'm assuming that the case on 179 handles regular User records?

File: src/sentry/models/actor.py [hunk 2]
```diff
@@ -179,17 +207,16 @@ def resolve_many(cls, actors: Sequence["ActorTuple"]) -> Sequence[Union["Team",
         return list(filter(None, [results.get((actor.type, actor.id)) for actor in actors]))
 
 
-def handle_actor_pre_save(instance, **kwargs):
+def handle_team_post_save(instance, **kwargs):
     # we want to create an actor if we don't have one
     if not instance.actor_id:
         instance.actor_id = Actor.objects.create(
-            type=ACTOR_TYPES[type(instance).__name__.lower()]
+            type=ACTOR_TYPES[type(instance).__name__.lower()],
```
Comment (markstory) [inline src/sentry/models/actor.py:214]:
```suggestion
            type=ACTOR_TYPES['team'],
```

I don't think we need to metaprogram this anymore.

File: src/sentry/models/team.py [hunk 3]
```diff
@@ -280,14 +285,22 @@ def transfer_to(self, organization):
...
                 cursor.execute("DELETE FROM sentry_team WHERE id = %s", [self.id])
+                cursor.execute("DELETE FROM sentry_actor WHERE team_id = %s", [new_team.id])
             finally:
                 cursor.close()
 
+            # Change whatever new_team's actor is to the one from the old team.
+            with transaction.atomic():
+                Actor.objects.filter(id=self.actor_id).update(team_id=new_team.id)
+                new_team.actor_id = self.actor_id
+                new_team.save()
```
Comment (markstory) [inline src/sentry/models/team.py:302]: Should we have a test for this as re-using the actor is something we could overlook in the future.

**Comment by corps** [comment]
I'm doing this the wrong way -- switch my plan up.

**Pull request closed without merging**